oedisi/tools/cli_tools.py

```diff
@@ -80,8 +102,183 @@ def build(target_directory, system, component_dict):
+        yaml.dump(service, f)
+
+
+def clone_broker(target_directory, yaml_file_path):
+    broker_folder = os.path.join(target_directory, 'broker')
+    if not os.path.exists(broker_folder):
+        os.makedirs(broker_folder)
+    copy_tree('./broker', broker_folder)
+    shutil.copy(yaml_file_path, broker_folder)
+    return
+
+def edit_docker_file(file_path, component):
+    edited_lines = []
+    with open(file_path, 'r') as f:
+        for line in f.readlines():
+            if line.startswith('RUN mkdir'):
+                edited_lines.append(f'RUN mkdir {component.name}\n')
+            elif line.startswith('COPY  *'):
+                edited_lines.append(f'COPY  * ./{component.name}\n')
+            elif line.startswith('WORKDIR'):
+                edited_lines.append(f'WORKDIR ./{component.name}\n')
+            elif line.startswith('EXPOSE'):
+                edited_lines.append(f'EXPOSE {component.port}/tcp\n')
+            elif line.startswith('CMD'):
+                a = f'CMD {["python", "server.py", component.host, str(component.port)]}\n'
+                a = a.replace("'", '"')
+                edited_lines.append(a)
+            else:
+                edited_lines.append(line)
+    with open(file_path, 'w') as f:
+        f.writelines(edited_lines)
+    pass
 
 
+def edit_docker_files(wiring_diagram:WiringDiagram, target_directory:str):
+    for component in wiring_diagram.components:
+        docker_path = os.path.join(target_directory, component.name, "Dockerfile")
+        try:
+            edit_docker_file(docker_path, component)
+        except Exception as e:
+            print(e)
+    ...
+
+
+def create_docker_compose_file(wiring_diagram:WiringDiagram, target_directory:str):
+    config = {"services": {}, "networks": {}}
+    
+    config['services'][f"{APP_NAME}_{BROKER_SERVICE}"] = {
+            "build": {
+                    "context": f'./broker/.'
+                },
+            "ports": [
+                    f'{BROKER_PORT}:{BROKER_PORT}'
+                ],
+            "networks": {
+                "custom-network" : {
+                        "ipv4_address": '10.5.0.2'
+                    } 
+                },
+            }
+    
+    for component in wiring_diagram.components:
+        config['services'][f"{APP_NAME}_{component.name}"] = {
+            "build": {
+                    "context": f'./{component.name}/.'
+                },
+            "ports": [
+                    f'{component.port}:{component.port}'
+                ],
+            "networks": {
+                "custom-network" : {
+                        "ipv4_address": component.host
+                    } 
+                },
+            }
+    
+    
+     
+    config["networks"] = {
+        "custom-network": {
+            "driver": "bridge",
+            "ipam": {
+                "config": [
+                    {
+                        "subnet": "10.5.0.0/16",
+                        "gateway": "10.5.0.1",
```

Comment:
I don't see a version in your docker compose, but it seems that gateway is no longer respected in docker-compose version 3: https://docs.docker.com/compose/compose-file/compose-file-v3/

If I understand this correctly, then this requires that all components use that subnet right?

Reply:
I will look into this